backend/app/src/main/java/com/moh/phlat/backend/service/SourceDataServiceImpl.java

```diff
@@ -35,81 +36,106 @@ public class SourceDataServiceImpl implements SourceDataService {
 	private SourceDataFilterSpecification specificationService = new SourceDataFilterSpecificationImpl();
 	
 	@Override
-	public List<SourceData> getSourceData(Long controlId, SourceDataFilterParams filterProcess) {
-		
-		Specification<SourceData> combinedSpecification;
+	public List<SourceData> getSourceData(Long controlId, SourceDataFilterParams filterSource, Pageable pageable) {
 		
 		try {
-			combinedSpecification = specificationService.buildSpecificationWhereEqual("controlTableId", controlId.toString());
+			return sourceDataRepository.findAll(buildSpecification(controlId, filterSource), pageable);
 		} catch (Exception e) {
 			logger.error("Error occured: {}", e.getMessage(), e);
 			
 			return new ArrayList();
 		}
+	}
+	
+	@Override
+	public Long countSourceData(Long controlId,  SourceDataFilterParams filterSource) {
+		try{
+			return sourceDataRepository.count(buildSpecification(controlId, filterSource));
+		} catch (Exception e) {
+			logger.error("Error occured: {}", e.getMessage(), e);
+			
+			return null;
```

Comment:
```suggestion
			return 0;
```
we might not need this method as commented in process data controller. If need, we should not be returning null for an integer return.

Reply:
After testing, changed the method in return an int and return a 0 if an error is sent.